### PR TITLE
Fix list filter on paste for RN mobile.

### PR DESCRIPTION
**Pull request by SergioEstevao**
## Description

It looks that the DOM implementation on RN does not directly support the spreadable operator on NodeList objects.
I converted the code to convert the NodeList to an Array instead of using the spreadable operator.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/blocks/src/api/raw-handling/list-reducer.js [hunk 1]
```diff
@@ -8,7 +8,9 @@ function isList( node ) {
 }
 
 function shallowTextContent( element ) {
-	return [ ...element.childNodes ]
+	// On mobile DOM implmentation NodeList is not spreadable so we need to convert it to an array.
```
Comment (ellatrix) [inline packages/blocks/src/api/raw-handling/list-reducer.js]: Does Array.from not work?

Comment (SergioEstevao) [reply packages/blocks/src/api/raw-handling/list-reducer.js]: Just tested and it works fine! Thanks for the suggestion.

File: packages/blocks/src/api/raw-handling/list-reducer.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ function isList( node ) {
...
+	const nodesArray = Array.from( element.childNodes );
```
Comment (ellatrix) [inline packages/blocks/src/api/raw-handling/list-reducer.js]: Why introduce the constant?

File: packages/blocks/src/api/raw-handling/list-reducer.js (hunk 1 above)
Comment (ellatrix) [inline packages/blocks/src/api/raw-handling/list-reducer.js]: The comment doesn't make entirely sense to me. The web implementation also needs to be converted to a real array, but it happens to be spreadable. In general we should always use `Array.from`. I believe we do in most places. Is this the only instance where we spread to convert it to a real array?

**Comment by ellatrix** [comment]
Maybe worth adjusting for consistency:

```
packages/edit-post/src/components/admin-notices/index.js
packages/dom/src/focusable.js
packages/components/src/higher-order/navigate-regions/index.js
packages/components/src/draggable/index.js
packages/components/src/disabled/test/index.js
packages/blocks/src/api/validation/index.js
```

File: packages/blocks/src/api/raw-handling/list-reducer.js (hunk 2 above)
Comment (SergioEstevao) [reply packages/blocks/src/api/raw-handling/list-reducer.js]: Ok I removed it and using directly the Array.from result

File: packages/blocks/src/api/raw-handling/list-reducer.js (hunk 1 above)
Comment (SergioEstevao) [reply packages/blocks/src/api/raw-handling/list-reducer.js]: I removed the comment. I don't know if this is the only instance were we spread the NodeList, I just tracked down the bug I was seeing when doing paste without format on the RN app and got to this method.

**Comment by SergioEstevao** [comment]
@ellatrix is this something related to this PR?

> Maybe worth adjusting for consistency:
> 
> packages/edit-post/src/components/admin-notices/index.js
> packages/dom/src/focusable.js
> packages/components/src/higher-order/navigate-regions/index.js
> packages/components/src/draggable/index.js
> packages/components/src/disabled/test/index.js
> packages/blocks/src/api/validation/index.js

**Comment by ellatrix** [comment]
Yes, those files all have cases where `NodeList` instances are spread.

**Comment by SergioEstevao** [comment]
Can I do those changes in a follow up PR targeting master?

This PR was to be used in GB-mobile PR to fix a very specific problem in
Copy/Paste without formatting.

**Comment by ellatrix** [comment]
Why can't it be done in one PR? But sure whatever you prefer. It just doesn't sounds like it would be changed then.

**Review by ellatrix — APPROVED** [review]

**Comment by SergioEstevao** [comment]
@ellatrix I changed the other files too, the only I coudn't find any instance of the problem was ```packages/blocks/src/api/validation/index.js``` i don't even have that file in my branch...

**Comment by gziolo** [comment]
There is one linting error which blocks this PR:
https://travis-ci.com/WordPress/gutenberg/jobs/239194179#L580
>    64:8  error  Variables should not be assigned until just prior its first reference. An early return statement may leave this variable unused  @wordpress/no-unused-vars-before-return

**Comment by ellatrix** [comment]
> @ellatrix I changed the other files too, the only I coudn't find any instance of the problem was `packages/blocks/src/api/validation/index.js` i don't even have that file in my branch...

There's only some documentation code that should be adjusted there. There's three doc blocks mentioning `[ ...document.querySelectorAll`.

**Comment by ellatrix** [comment]
Thanks for adjusting all the other instances!

**Comment by ellatrix** [comment]
@SergioEstevao I'm a bit confused... why is this not merged into `master`?

**Comment by gziolo** [comment]
It will be merged soon, there was `rnmobile/master` branch created because refactor applied to core entities broke mobile.

**Pull request merged**